lti_consumer/utils.py

```diff
@@ -26,22 +26,36 @@ def _(text):
     return text
 
 
-def get_lms_base():
+def get_lti_api_base():
     """
-    Returns LMS base url to be used as issuer on OAuth2 flows
-    and in various LTI URLs. For local testing it is often necessary
-    to override the normal LMS base with a proxy such as ngrok, use
-    the setting LTI_LMS_BASE_URL_OVERRIDE in your LMS settings if
-    necessary.
+    Returns base url to be used as issuer on OAuth2 flows
+    and in various LTI API calls.
 
     TODO: This needs to be improved and account for Open edX sites and
     organizations.
     One possible improvement is to use `contentstore.get_lms_link_for_item`
     and strip the base domain name.
     """
-    if hasattr(settings, 'LTI_LMS_BASE_URL_OVERRIDE'):
-        return settings.LTI_LMS_BASE_URL_OVERRIDE
+    if hasattr(settings, 'LTI_API_BASE_URL'):
+        return settings.LTI_API_BASE_URL
+    elif hasattr(settings, 'LTI_VIEW_BASE_URL'):
+        return settings.LTI_VIEW_BASE_URL
     else:
+        # TODO: we should move away from supporting this setting as it conflicts with usage
```

Comment:
Since this lib is still installed as an xblock I didn't want to go removing support for this just yet. That said, we do need a new setting because we can't hijack the use of LMS_ROOT_URL if this is running outside the LMS. This should point at the API for the current runtime and LMS_ROOT_URL is likely already pointing at the LMS.

Reply:
I wouldn't be surprised if we left this in forever - it's just too useful for the default LTI running out of the LMS case. We only need it to be different in dev and test and new IDAs

Reply:
my gut is this can't go anywhere unless we break out the entire xblock part into it's own codebase. So agreed, not any time soon.